tests/end_to_end/arg-temp/test.s

```diff
@@ -0,0 +1,5 @@
+glabel test
+sw $zero, ($sp)
+move $v0, $a0
+jr $ra
+nop
```

Comment:
This is a really interesting test case -- did you add this because you have a plan to separate out `$a0` from `0($sp)` in a future PR?

Reply:
This test case tests that temps for arguments and prevent_later_uses on arguments works correctly: before this PR, the output wouldn't include temp_a0 and thus be incorrect. With the PR there's a temp for a0 and the store to 0(sp) marks it as forced.